neural/neural_discern_run_script.py

```diff
@@ -0,0 +1,350 @@
+    # instance variable for computed articles reprs
+    proc_articles_repr = ReaderWriter.read_data(os.path.join(data_dir, 'processor_articles_repr.pkl'))
+    proc_config = ReaderWriter.read_data(os.path.join(data_dir, 'processor_config.pkl'))
+
+    return q_partitions, docs_data_tensor, proc_articles_dict, proc_articles_repr, proc_config
+
+
+def build_doc_partitions(questions, proc_articles_repr, proc_articles_dict, proc_config, docs_data_tensor, q_partitions,
+                         validate=False):
+    processor = DataDictProcessor(proc_config)
+    # reassign updated articles_dict and articles_repr to processor instance
+    processor.set_instance_attr(proc_articles_repr, proc_articles_dict, proc_config)
+
+    # turn the list of ids in q_partitions into dataset using PartitionDataTensor
+    q_docpartitions = {}
+    for question in questions:
+        q_docpartitions.update(generate_docpartition_per_question(docs_data_tensor, q_partitions, question))
+
+    if validate:
+        validate_q_docpartitions(q_docpartitions, q_partitions)
+
+    # add class weights in q_docpartitions
+    # we will use the weights to emphasize less occuring class during training phase
+    compute_class_weights_per_fold_(q_docpartitions)
+
+    return q_docpartitions
+
+
+def load_biobert_model(biobert_pth, device):
+    """Read saved state dict for biobert model on disk
+
+    Args:
+        biobert_pth: str, folder path where model state dictionary and config file are saved
+    """
+    bert_config_file = os.path.join(biobert_pth, 'bert_config.json')
+    config = BertConfig.from_json_file(bert_config_file)
+    print("Building PyTorch model from configuration: {}".format(str(config)))
+    model = BertForPreTraining(config)
+    model.load_state_dict(torch.load(os.path.join(biobert_pth, 'biobert_statedict.pkl'), map_location=device))
+    return model
+
+
+def write_sents_embeddings(directory, bertmodel, sents_embed_dir_name, docs_data_tensor):
+    # === Generate sents embedding ===
+    # load BertModel
+
+    # define BertEmbedder
+    bert_config = {'bert_train_flag': False,
+                   'bert_all_output': False}
+    bertembeder = BertEmbedder(bertmodel, bert_config)
+    sents_embed_dir = create_directory(sents_embed_dir_name, directory)
+    fdtype = torch.float32
+
+    # generate and dump bert embedding for the tokens inside the specificed embedding directory
+    bert_proc_docs = generate_sents_embeds_from_docs(docs_data_tensor, bertembeder, sents_embed_dir, fdtype)
+    ReaderWriter.dump_data(bert_proc_docs, os.path.join(sents_embed_dir, 'bert_proc_docs.pkl'))
+
+
+def read_sents_embeddings(directory, sents_embed_dir_name):
+    ''' read the 3-dimensional doc tensors
+    doc x sent x embedding'''
+
+    sents_embed_dir = create_directory(sents_embed_dir_name, directory)
+    bert_proc_docs = ReaderWriter.read_data(os.path.join(sents_embed_dir, 'bert_proc_docs.pkl'))
```

Comment:
I think bert_proc_docs is a dictionary {`article_id`:`path_for_embedding_tensor_on_disk`}.

Reply:
Oh I see now. I'm removing the whole function because `bert_proc_docs` isn't used anywhere in this script.